acceptance/suites/puppet3_tests/020_backwards_compat/test.rb

```diff
@@ -0,0 +1,118 @@
+# require 'test/unit/assertions'
+# require 'puppet/acceptance/module_utils'
+
+test_name "3.x agent running against 4.x master"
+
+studio = "/tmp/simmons-studio-#{Process.pid}"
+
+teardown do
+  on(agents, "rm -rf #{studio}")
+end
+
+step "Install modules" do
+  simmons_version = '0.2.0'
+  on(master, puppet("module install nwolfe-simmons --version #{simmons_version}"))
+  on(master, puppet("module list"))
+end
+
+step "Configure file serving" do
+  fileserverconf = on(master, puppet("config print fileserverconfig")).stdout.chomp
+  create_remote_file(master, fileserverconf, <<FILESERVERCONF)
+[simmons_custom_mount_point]
+path /etc/puppetlabs/code/environments/production/modules/simmons/mount-point-files
+allow *
+FILESERVERCONF
+  on(master, "chmod 644 #{fileserverconf}")
+end
+
+step "Disable auth.conf" do
+  authconf = on(master, puppet("config print rest_authconfig")).stdout.chomp
+  create_remote_file(master, authconf, <<AUTHCONF)
+path /
+auth any
+allow *
+AUTHCONF
+  on(master, "chmod 644 #{authconf}")
+end
+
+step "Perform agent upgrade steps" do
+  step "Enable structured facts" do
+    agents.each do |agent|
+      on(agent, puppet("config set stringify_facts false --section agent"))
+    end
+  end
+end
+
+step "Configure site.pp" do
+  sitepp = '/etc/puppetlabs/code/environments/production/manifests/site.pp'
+  create_remote_file(master, sitepp, <<SITEPP)
+class { 'simmons':
+  studio => "#{studio}",
+}
+SITEPP
+  on(master, "chmod 644 #{sitepp}")
+end
+
+agent = master
+
+step "Run agents" do
+  with_puppet_running_on(master, {}) do
```

Comment:
Yeah this was what I was referring to today in standup, about quirks that I'll need to work out once I actually run the 3.x agent against the 4.x master.  Right now, I have the 4.x agent running against the 4.x master for development, so that happens to avoid this issue since it's re-using the master's certificate.

Yeah I'll have to turn autosigning probably